clients/da-vinci-client/src/main/java/com/linkedin/davinci/blobtransfer/client/P2PFileTransferClientHandler.java

```diff
@@ -176,6 +177,16 @@ protected void channelRead0(ChannelHandlerContext ctx, HttpObject msg) throws Ex
                   + receivedFileChecksum);
         }
 
+        // One file transfer completed, reset the state for the next file transfer
+        try {
+          outputFileChannel.close();
```

Comment:
you can just wrap it with try-with-resource directly above.

Reply:
Yeah this is fine to me, as long as we manage to close the channel at least once.